### PR TITLE
Implement Vue Chat example

### PR DESCRIPTION
**Pull request by niallobrien**
## Description

Implements Vue.js example which will supplement a later Vue.js framework guide at http://docs.feathersjs.com/frameworks/vuejs.html

## Solution

Demonstrates template usage, components, props, slots (transclusion) and filters.
Components: ChatApp, UserList, MessageList, Message, ComposeMessage

## Tested Browsers

* Chrome v.49 (Mac OS X 10.11.3) [Macbook Air].
* Firefox v.44.0.2 (Mac OS X 10.11.3) [Macbook Air]

### CODE REVIEW

**Comment by ekryski** [comment]
Ooooooh Yeeaaah. I'll check this out this evening or tomorrow.

**Comment by daffl** [comment]
Looks great! 👍 from me.

**Comment by ekryski** [comment]
:shipit:

File: public/vue-chat/app.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
+// A placeholder image if the user does not have one
+const PLACEHOLDER = '/placeholder.png';
```
Comment (ekryski) [inline public/vue-chat/app.js]: Ah shoot. I guess the only thing we need to do is change this to the new placeholder image.

Comment (ekryski) [inline public/vue-chat/app.js]: Fixed it.

**Comment by niallobrien** [comment]
Thanks guys, the only thing I wasn't sure of was for example fetching the users & messages in the chat-app component (parent component) and passing the props down to the respective components much like in the React example. In this Vue example, I thought it made more sense to fetch the users within the user component and fetch the messages within the messages component. But to demonstrate the use of props in Vue, I nested a message component within the message-list component.
Just wondering what your thoughts are on this approach? Thanks.

**Comment by ekryski** [comment]
@niallobrien I think this is totally fine. Nice work and thanks again!

**Comment by niallobrien** [comment]
Thanks @ekryski :)

**Pull request merged**